src/cli/cmd/apply/mod.rs

```diff
@@ -0,0 +1,275 @@
+            System::NixOs(nixos) => Box::new(nixos),
+            System::NixDarwin(nix_darwin) => Box::new(nix_darwin),
+        };
+
+        let output_ref = {
+            parse_output_ref(
+                &self.frontend_addr,
+                applyer.get_ref(),
+                &applyer.default_ref(),
+            )?
+        };
+
+        tracing::info!("Resolving {}", output_ref);
+
+        let resolved_path = FlakeHubClient::resolve(self.api_addr.as_ref(), &output_ref).await?;
+
+        tracing::debug!(
+            "Successfully resolved reference {} to path {}",
+            &output_ref,
+            &resolved_path.store_path
+        );
+
+        let (profile_path, _tempdir) = apply_path_to_profile(
+            applyer.profile_path(),
+            &resolved_path.store_path,
+            applyer.requires_root(),
+        )
+        .await?;
+
+        let script_path = profile_path.join(applyer.relative_path());
+
+        run_script(
+            script_path,
+            applyer.action(),
+            &applyer
+                .relative_path()
+                .file_name()
+                .expect("The apply type should absolutely have a file name.")
+                .to_string_lossy(),
+        )
+        .await?;
+
+        Ok(ExitCode::SUCCESS)
+    }
+}
+
+// This function enables you to provide simplified paths:
+//
+// fh apply nixos omnicorp/systems/0.1
+//
+// Here, `omnicorp/systems/0.1` resolves to `omnicorp/systems/0.1#nixosConfigurations.$(hostname)`.
+// If you need to apply a configuration at a path that doesn't conform to this pattern, you
+// can still provide an explicit path.
+fn parse_output_ref(
+    frontend_addr: &url::Url,
+    output_ref: &str,
+    default_path: &str,
+) -> Result<super::FlakeOutputRef, FhError> {
+    let with_default_output_path = match output_ref.split('#').collect::<Vec<_>>()[..] {
+        [_release, _output_path] => output_ref.to_string(),
+        [_release] => format!("{}#{}", output_ref, default_path),
+        _ => return Err(FhError::MalformedOutputRef(output_ref.to_string())),
+    };
+
+    let output_ref =
+        super::parse_flake_output_ref(frontend_addr, &with_default_output_path)?.to_string();
+
+    let parsed = super::parse_release_ref(&output_ref)?;
+
+    parsed.try_into()
+}
+
+async fn run_script(
+    script_path: PathBuf,
+    action: Option<String>,
+    script_name: &str,
+) -> Result<(), FhError> {
+    tracing::debug!(
+        "Checking for {} script at {}",
+        script_name,
+        &script_path.display().to_string(),
+    );
+
+    if script_path.exists() && script_path.is_file() {
+        tracing::debug!(
+            "Found {} script at {}",
+            script_name,
+            &script_path.display().to_string(),
+        );
+
+        if let Ok(script_path_metadata) = tokio::fs::metadata(&script_path).await {
+            let permissions = script_path_metadata.permissions();
+            if permissions.mode() & 0o111 != 0 {
```

Comment:
Can I get a brief comment on what this is checking? And if the check fails, can we have some sort of logging to let the user know that something was in an unexpected state?